src/main/java/net/pms/util/ImagesUtil.java

```diff
@@ -92,57 +158,1394 @@ public static void auto(InputStream input, OutputStream output, int exifOrientat
+	 * </b>
+	 * @param inputByteArray the source image in a supported format.
+	 * @param inputImage the source {@link Image}.
+	 * @param inputStream the source image in a supported format.
+	 * @param width the new width or 0 to disable scaling.
+	 * @param height the new height or 0 to disable scaling.
+	 * @param scaleType the {@link ScaleType} to use when scaling.
+	 * @param outputFormat the {@link ImageFormat} to convert to or
+	 *                     {@link ImageFormat#SOURCE} to preserve source
+	 *                     format. Overridden by {@code outputProfile}.
+	 * @param outputProfile the {@link DLNAImageProfile} to convert to. This
+	 *                      overrides {@code outputFormat}.
+	 * @param updateMetadata whether or not new metadata should be updated
+	 *                       after image transformation. This should only be
+	 *                       disabled if the output image won't be kept/reused.
+	 * @param dlnaCompliant whether or not the output image should be
+	 *                      restricted to DLNA compliance. This also means that
+	 *                      the output can be safely cast to {@link DLNAImage}.
+	 * @param dlnaThumbnail whether or not the output image should be
+	 *                      restricted to DLNA thumbnail compliance. This also
+	 *                      means that the output can be safely cast to
+	 *                      {@link DLNAThumbnail}.
+	 * @param padToSize whether padding should be used if source aspect doesn't
+	 *                  match target aspect.
+	 * @return The scaled and/or converted image or {@code null} if the source
+	 *         is {@code null}.
+	 * @throws IOException if the operation fails.
+	 */
+	protected static Image transcodeImage(
+		byte[] inputByteArray,
+		Image inputImage,
+		InputStream inputStream,
+		int width,
+		int height,
+		ScaleType scaleType,
+		ImageFormat outputFormat,
+		DLNAImageProfile outputProfile,
+		boolean updateMetadata,
+		boolean dlnaCompliant,
+		boolean dlnaThumbnail,
+		boolean padToSize
+	) throws IOException {
+		if (inputByteArray == null && inputStream == null && inputImage == null) {
+			return null;
+		}
+		if (
+			(inputByteArray != null & inputImage != null) ||
+			(inputByteArray != null & inputStream != null) ||
+			(inputImage != null & inputStream != null)
+		) {
+			throw new IllegalArgumentException("Use either inputByteArray, inputImage or inputStream");
+		}
+		ImageIO.setUseCache(false);
+
+		if (inputImage != null) {
+			inputByteArray = inputImage.getBytes(false);
+		} else if (inputStream != null) {
+			inputByteArray = ImagesUtil.toByteArray(inputStream);
+		}
 
-				tf = jpegmeta.findEXIFValue(TiffTagConstants.TIFF_TAG_ORIENTATION);
-				if (tf != null) {
-					media.setOrientation(tf.getIntValue());
+		// outputProfile overrides outputFormat
+		if (outputProfile != null) {
+			if (dlnaThumbnail && outputProfile.equals(DLNAImageProfile.GIF_LRG)) {
+				outputProfile = DLNAImageProfile.JPEG_LRG;
```

Comment:
This is just a fail-safe, no caller should ever ask for a DLNA thumbnail with profile ```GIF_LRG```. I'm simply choosing the "corresponding" JPEG profile. Since ```GIF_LRG``` has a maximum of 1600 x 1200, ```JPEG_LRG``` and ```JPEG_RES_H_V``` are the only two profiles big enough to hold "any" ```GIF_LRG``` - but I'm not expecting this to ever be used. I could just has well throw an ```Exception``` saying that's an illegal combination, but I generally try to make the code as forgiving as possible without compromising its integrity.